simulation/traffic_simulator/include/traffic_simulator/entity/entity_manager.hpp

```diff
@@ -156,30 +176,72 @@ class EntityManager
       configuration.lanelet2_map_path(), getOrigin(*node))),
     markers_raw_(hdmap_utils_ptr_->generateMarker()),
-    traffic_light_manager_ptr_(makeTrafficLightManager(hdmap_utils_ptr_, node))
+    conventional_traffic_light_manager_ptr_(
+      makeConventionalTrafficLightManager(hdmap_utils_ptr_, node)),
+    v2i_traffic_light_manager_ptr_(makeV2ITrafficLightManager(hdmap_utils_ptr_, node))
   {
     updateHdmapMarker();
   }
 
   ~EntityManager() = default;
 
-public:
-  template <typename... Ts>
-  auto getTrafficLight(Ts &&... xs) const -> decltype(auto)
+private:
+  struct TrafficLightManager
   {
-    return traffic_light_manager_ptr_->getTrafficLight(std::forward<decltype(xs)>(xs)...);
-  }
+    std::shared_ptr<TrafficLightManagerBase> conventional_traffic_light_manager_ptr_;
+    std::shared_ptr<TrafficLightManagerBase> v2i_traffic_light_manager_ptr_;
+  } traffic_light_manager;
 
-  auto getTrafficLights() const -> decltype(auto)
+public:
+#define FORWARD_GETTER_TO_TRAFFIC_LIGHT_MANAGER(NAME)                 \
+  template <typename... Ts>                                           \
+  decltype(auto) getConventional##NAME(Ts &&... xs) const             \
+  {                                                                   \
+    return conventional_traffic_light_manager_ptr_->get##NAME(xs...); \
+  }                                                                   \
+  static_assert(true, "");                                            \
+  template <typename... Ts>                                           \
+  decltype(auto) getV2I##NAME(Ts &&... xs) const                      \
+  {                                                                   \
+    return v2i_traffic_light_manager_ptr_->get##NAME(xs...);          \
+  }                                                                   \
+  static_assert(true, "")
+
+  //  auto getConventionalTrafficLights(std::int64_t lanelet_id) const {
+  //    return conventional_traffic_light_manager_ptr_->getTrafficLights(lanelet_id);
+  //  }
+  //
+  //  auto getV2ITrafficLights(std::int64_t lanelet_id) const {
+  //    return v2i_traffic_light_manager_ptr_->getTrafficLights(lanelet_id);
+  //  }
+  //
+  //  auto getConventionalTrafficLights() const {
+  //    return conventional_traffic_light_manager_ptr_->getTrafficLights();
+  //  }
+  //
+  //  auto getV2ITrafficLights() const {
+  //    return v2i_traffic_light_manager_ptr_->getTrafficLights();
+  //  }
+  //
+  //  auto getConventionalTrafficLight(std::int64_t lanelet_id) const {
+  //    return conventional_traffic_light_manager_ptr_->getTrafficLight(lanelet_id);
+  //  }
+  //
+  //  auto getV2ITrafficLight(std::int64_t lanelet_id) const {
+  //    return v2i_traffic_light_manager_ptr_->getTrafficLight(lanelet_id);
+  //  }
+
+  FORWARD_GETTER_TO_TRAFFIC_LIGHT_MANAGER(TrafficLights);
+  FORWARD_GETTER_TO_TRAFFIC_LIGHT_MANAGER(TrafficLight);
+
+  auto updateConventionalTrafficLightsPublishRate(double rate) -> void
```

Comment:
In the existing code of traffic_simulator, "update" is a verb assigned exclusively to a subroutine called every frame. So avoid giving the word "update" to functions that behave differently.

Reply:
use `apply` instead of `update` in https://github.com/tier4/scenario_simulator_v2/pull/1004/commits/9b7a262597ea86ff02ce771a0db7095d19dd3cfa

Reply:
use `reset` instead of `update` in https://github.com/tier4/scenario_simulator_v2/pull/1004/commits/76374fcd98686c413b3e90e8851ddfa9fb78b5d2